hw_diag/utilities/quectel.py

```diff
@@ -150,121 +168,151 @@ def _do_upgrade(desired_fw_version: str) -> bool:
     return upgrade_successful
 
 
 def is_internet_accessible(wait_time: int = INTERNET_MAX_WAIT_TIME) -> bool:
-    '''wait for wait_time seconds for modem to achieve connectivity'''
+    """wait for wait_time seconds for modem to achieve connectivity"""
     try:
         delay_interval = 5
-        max_tries = wait_time/delay_interval
-        retry_call(urllib.request.urlopen,
-                   fkwargs={'url': 'https://google.com', 'timeout': 3},
-                   tries=max_tries, delay=delay_interval, logger=logging)
+        max_tries = wait_time / delay_interval
+        retry_call(
+            urllib.request.urlopen,
+            fkwargs={"url": "https://google.com", "timeout": 3},
+            tries=max_tries,
+            delay=delay_interval,
+            logger=logging,
+        )
         return True
     except Exception as e:
-        logging.info(f'internet not accessible: {e}')
+        logging.info(f"internet not accessible: {e}")
         return False
 
 
-def update_setting_with_rollback(getter: str, setter: str, desired_value: str) -> bool:
-    modem = find_eg25g_modem()
-    if not modem:
-        logging.debug('EG25G modem not found')
-        return False
+def call_method(obj, method, *args, **kwargs):
+    return getattr(obj, method)(*args, **kwargs)
+
+
+def setting_needs_update(
+    modem: Modem, getter: str, setter: str, desired_value: str
+) -> Tuple[bool, str]:
+    """return a tuple containing
+    true if settings needs to be updated, false otherwise
+    and current value of the setting"""
+    old_value = call_method(modem, getter)
+    if at_max_retries(setter, SETTINGS_MAX_RETRIES):
+        return False, old_value
+
+    if old_value == desired_value:
+        logging.info(f"Setting is already correct {old_value}")
+        return False, old_value
+    logging.info(f"actual value : {old_value} needs to be: {desired_value}")
+    return True, old_value
 
-    getter_method = getattr(modem, getter)
-    setter_method = getattr(modem, setter)
 
+def update_setting(modem: Modem, getter: str, setter: str, desired_value: str) -> bool:
+    """returns true if the setting was correctly updated, false otherwise"""
     try:
-        # this settings has been tried unsuccessfully too many times
-        if at_max_retries(setter, SETTINGS_MAX_RETRIES):
+        call_method(modem, setter, desired_value)
+        new_value = call_method(modem, getter)
+        if desired_value != new_value:
+            logging.error("the value didn't set correctly")
             return False
+        return True
+    except Exception as e:
+        logging.info(f"failed to update setting {setter} to {desired_value}")
+        logging.error(f"error: {e}")
+        return False
 
-        old_value = getter_method()
-        if old_value == desired_value:
-            logging.info(f"Setting is already correct {old_value}")
+
+def update_setting_with_rollback(getter: str, setter: str, desired_value: str) -> bool:
```

Comment:
`getter` and `setter` are confusing variable names. I associate them with instance functions but these are strings.

Reply:
Ah OK, now I see how this is used in the code and it makes sense. It would be helpful to add documentation here that explains.

Reply:
added documentation